File: Scala/src/main/scala/BinaryNumbers.scala
package main.scala

object BinaryNumbers {
  def main(args: Array[String]){
    val n = scala.io.StdIn.readInt()
    val binary = Integer.toString(n,2)
    var set = 0
    var max = 0
    for(i <- 0 to binary.length -1){
      if(binary.charAt(i) == '1'){
        set = set + 1
        if(max < set){
          max = set
        }
      } else {
        set = 0
      }

    }
    println(max)
  }
}

File: Java/src/com/hackerrank/daysofcode/BinaryNumbers.java
package com.hackerrank.daysofcode;
import java.io.*;
import java.util.*;

public class BinaryNumbers {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int input = scan.nextInt();
        String binary =  Integer.toString(input,2);
        int set = 0, max = 0;
        for(int i = 0; i < binary.length(); i++){
            if(binary.charAt(i) == '1'){
                set++;
                if(max < set){
                    max = set;
                }
            } else {
                set = 0;
                continue;
            }
        }
        System.out.println(max);
    }
}

